tests/extras/datasets/api/test_api_dataset.py

```diff
@@ -17,41 +17,43 @@
 
 TEST_PARAMS = {"param": "value"}
 TEST_URL_WITH_PARAMS = TEST_URL + "?param=value"
-
+TEST_METHOD = "GET"
 TEST_HEADERS = {"key": "value"}
 
 
-@pytest.mark.parametrize("method", POSSIBLE_METHODS)
 class TestAPIDataSet:
-    @pytest.fixture
-    def requests_mocker(self):
-        with requests_mock.Mocker() as mock:
-            yield mock
-
-    def test_successfully_load_with_response(self, requests_mocker, method):
+    @pytest.mark.parametrize("method", POSSIBLE_METHODS)
+    def test_successfully_load_with_response(self, requests_mock, method):
         api_data_set = APIDataSet(
-            url=TEST_URL, method=method, params=TEST_PARAMS, headers=TEST_HEADERS
+            url=TEST_URL, method=method, load_args={"params": TEST_PARAMS}
         )
-        requests_mocker.register_uri(
-            method,
-            TEST_URL_WITH_PARAMS,
-            headers=TEST_HEADERS,
-            text=TEST_TEXT_RESPONSE_DATA,
+        requests_mock.register_uri(
+            method, TEST_URL_WITH_PARAMS, text=TEST_TEXT_RESPONSE_DATA
         )
 
         response = api_data_set.load()
         assert isinstance(response, requests.Response)
         assert response.text == TEST_TEXT_RESPONSE_DATA
 
-    def test_successful_json_load_with_response(self, requests_mocker, method):
+    def test_headers_in_request(self, requests_mock):
+        api_data_set = APIDataSet(
+            url=TEST_URL, method=TEST_METHOD, load_args={"headers": TEST_HEADERS}
+        )
+        requests_mock.register_uri(TEST_METHOD, TEST_URL, headers={"pan": "cake"})
+
+        response = api_data_set.load()
+
+        assert response.request.headers["key"] == "value"
+        assert response.headers["pan"] == "cake"
```

Comment:
Please can you explain these lines? Is the first line the request and the second the response? Why isn't it `key=pan` and `value=cake`? 🤔

Reply:
There is no difference between `requests_mock.last_request` and `response.request`. For clarity, I have changed all usages into `response.request`

Reply:
> Please can you explain these lines? Is the first line the request and the second the response? Why isn't it key=pan and value=cake? 🤔

Requests headers (a bit of information you send along with the request) are not the same as response headers (a bit of information sent back by the server). In the above example, the requests headers are literally `{"key": "value"}`. The response headers are `{"pan": "cake"}`.

Reply:
Got it, thank you!